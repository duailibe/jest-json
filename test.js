"use strict";

require(".");

describe("toMatchJSON", () => {
  const json = JSON.stringify({ foo: "bar", spam: "eggs" });

  test("matches", () => {
    expect(json).toMatchJSON({ foo: "bar", spam: "eggs" });
    expect(json).toMatchJSON({ spam: "eggs", foo: "bar" });
  });

  test(".not doesn't match", () => {
    expect(json).not.toMatchJSON({ foo: "baz", spam: "eggs" });
  });

  test("assertion error has a nice output", () => {
    expect(() =>
      expect(json).toMatchJSON({ foo: "baz" })
    ).toThrowErrorMatchingSnapshot();
    expect(() =>
      expect(json).not.toMatchJSON({ foo: "bar", spam: "eggs" })
    ).toThrowErrorMatchingSnapshot();
    expect(() =>
      expect(json).not.toMatchJSON({ foo: expect.anything(), spam: "eggs" })
    ).toThrowErrorMatchingSnapshot();
  });

  test("throws on invalid JSON", () => {
    expect(() => expect(null).toMatchJSON()).toThrowErrorMatchingSnapshot();
    expect(() => expect("").toMatchJSON()).toThrowErrorMatchingSnapshot();
    expect(() => expect("fals").toMatchJSON()).toThrowErrorMatchingSnapshot();
    expect(() => expect("falsr").toMatchJSON()).toThrowErrorMatchingSnapshot();
    expect(() => expect("fals'").toMatchJSON()).toThrowErrorMatchingSnapshot();
    expect(() => expect("fals9").toMatchJSON()).toThrowErrorMatchingSnapshot();
  });
});

describe("jsonMatching", () => {
  test("matches object", () => {
    expect(JSON.stringify({ foo: "bar" })).toEqual(
      expect.jsonMatching({
        foo: expect.any(String),
      })
    );

    expect(JSON.stringify({ foo: "bar", bar: "baz" })).toEqual(
      expect.jsonMatching(expect.objectContaining({ foo: "bar" }))
    );
  });

  test("matches array", () => {
    expect(JSON.stringify(["foo", "bar"])).toEqual(
      expect.jsonMatching(expect.arrayContaining(["bar", "foo"]))
    );
  });

  test("works inside arrayContaining()", () => {
    expect([1, JSON.stringify({ foo: "bar" })]).toEqual(
      expect.arrayContaining([expect.jsonMatching({ foo: "bar" })])
    );
  });
});
