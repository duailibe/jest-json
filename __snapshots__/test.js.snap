// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`toMatchJSON throws on invalid JSON 1`] = `
expect(string).toMatchJSON(expected)

string value must be a string.
Received has value: null
`;

exports[`toMatchJSON throws on invalid JSON 2`] = `
expect(string).toMatchJSON(expected)

string value must be a valid JSON.
Received:
  ""
`;

exports[`toMatchJSON throws on invalid JSON 3`] = `
expect(string).toMatchJSON(expected)

string value must be a valid JSON.
Received:
  fals
       ^
Unexpected end of string

`;

exports[`toMatchJSON throws on invalid JSON 4`] = `
expect(string).toMatchJSON(expected)

string value must be a valid JSON.
Received:
  falsr
       ^
Unexpected token: r
`;

exports[`toMatchJSON throws on invalid JSON 5`] = `
expect(string).toMatchJSON(expected)

string value must be a valid JSON.
Received:
  fals'
       ^
Unexpected token: '
`;

exports[`toMatchJSON throws on invalid JSON 6`] = `
expect(string).toMatchJSON(expected)

string value must be a valid JSON.
Received:
  fals9
       ^
Unexpected number: 9
`;
